src/elements/content-uploader/Footer.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { Button } from '@box/blueprint-web';
+import { ERROR_CODE_UPLOAD_FILE_LIMIT } from '../../constants';
+
+import messages from '../common/messages';
+
+import './Footer.scss';
+
+export interface FooterProps {
+    errorCode?: string;
+    fileLimit: number;
+    hasFiles: boolean;
+    isDone: boolean;
+    isLoading: boolean;
+    onCancel: () => void;
+    onClose?: () => void;
+    onUpload: () => void;
+}
+
+const Footer = ({ isLoading, hasFiles, errorCode, onCancel, onClose, onUpload, fileLimit, isDone }: FooterProps) => {
+    const intl = useIntl();
+    const isCloseButtonDisabled = hasFiles;
+    const isCancelButtonDisabled = !hasFiles || isDone;
+    const isUploadButtonDisabled = !hasFiles;
+
+    let message;
+    switch (errorCode) {
+        case ERROR_CODE_UPLOAD_FILE_LIMIT:
+            message = <FormattedMessage {...messages.uploadErrorTooManyFiles} values={{ fileLimit }} />;
+            break;
+        default:
+        // ignore
```

Comment:
is this comment needed? If so, can we give more info

Reply:
probably not, left over from the conversion from js, removing

Reply:
actually replacing the switch to an if